App.tsx

```diff
@@ -1,25 +1,50 @@
-import React from "react";
-import { Provider } from "react-redux";
-import { store } from "./src/redux/store"
-import { SafeAreaProvider } from 'react-native-safe-area-context';
+import React, { useEffect } from "react";
+
+import { Provider, useDispatch } from "react-redux";
+import { store } from "./src/redux/store";
+import { SafeAreaProvider } from "react-native-safe-area-context";
 import getCacheLoadData from "./src/hooks/onStartFetchData";
-import useColorScheme from "./src/hooks/useColorScheme"
-import Navigation from "./src/navigation"
+import useColorScheme from "./src/hooks/useColorScheme";
+import Navigation from "./src/navigation";
+import { firebaseApp } from "./src/constants/Firebase";
 
-export default function App() {
+function App() {
   const isLoadingComplete = getCacheLoadData();
   const colorScheme = useColorScheme();
 
+  const dispatch = useDispatch();
+
+  // If already logged in - set user
+  useEffect(() => {
+    firebaseApp.auth().onAuthStateChanged((user) => {
+      const currentUser = {
+        id: user?.uid,
+        username: user?.email,
+        image_url: user?.photoURL,
+      };
+      if (user !== null) {
+        dispatch({ type: "LOGIN_USER", payload: currentUser });
```

Comment:
Can you please import the setUser() method from our actions (index.ts) file, instead of dispatching an inline object.

Reply:
Additionally, I may change this later so that we change the initial state of UserState if we can authenticate the user from mobile storage. This way we take one step away and simply START with the User already logged in from the beginning of redux store initialization